esx_service/utils/vmdk_utils.py

```diff
@@ -73,10 +73,11 @@ def init_datastoreCache(force=False):
     from local ESX host. force=True will force it to ignore current cache
     and force init
     """
+    global datastores
     logging.debug("init_datastoreCache:  %s", datastores)
 
     with lockManager.get_lock("init_datastoreCache"):
-        global datastores
+
```

Comment:
as @govint has pointed out, we need to have references of datastores under the lock.
Need to move line 76 and 77 under with... statement.